libraries/botbuilder-dialogs/tests/waterfallDialog.test.js

```diff
@@ -634,4 +634,61 @@ describe('WaterfallDialog', function () {
         }, DialogReason.cancelCalled);
         assert(trackEventCalled, 'trackEvent was never called.');
     });
+
+    it('should waterfall parent be equal to Component Dialog.', (done) => {
+        const conversationState = new ConversationState(new MemoryStorage());
+        const dialogState = conversationState.createProperty('dialog');
+
+        const waterfall = new WaterfallDialog('waterfall', [
+            step => ({ status: DialogTurnStatus.complete, result: step })
+        ]);
+
+        const component = new ComponentDialog('composite')
+        component.addDialog(waterfall)
+
+        const dialogs = new DialogSet(dialogState);
+        dialogs.add(component);
+
+        const adapter = new TestAdapter(async turnContext => {
+            const dc = await dialogs.createContext(turnContext);
+
+            const results = await dc.beginDialog('composite');
+
+            if (results.status === DialogTurnStatus.complete) {
+                const waterfallParent = results.result.parent.parent;
+
+                assert.strictEqual(waterfallParent.dialogs.dialogs['composite'], component, 'WaterfallDialog "waterfall" parent should be equal to ComponentDialog "composite".');
+            }
+
+            done();
```

Comment:
If the result.status is not DialogTurnStatus.complete, then this test passes because done() is called.